### ISSUE TITLE
aws_batch_job_definition invalid or unknown key: timeout

### ISSUE THREAD

**Issue by CaseyPeters** [terraform-providers/terraform-provider-aws#4756]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.7 
Provider.aws v1.13.0

### Affected Resource(s)

* aws_batch_job_definition

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
resource "aws_batch_compute_environment" "staging-platform" {
    compute_environment_name = "staging-platform"
    service_role = "arn:aws:iam::442998785547:role/service-role/AWSBatchServiceRole"
    type = "MANAGED"
    compute_resources {
        instance_role = "arn:aws:iam::442998785547:instance-profile/campspot-instance-role"
        instance_type = ["optimal"]
        max_vcpus = 32
        min_vcpus = 0
        security_group_ids = ["${aws_security_group.staging-instance.id}"]
        subnets = ["${aws_subnet.subnet-staging-1a.id}", "${aws_subnet.subnet-staging-1c.id}"]
        type = "EC2"
        tags {
            Name = "staging-batch-jobs"
        }
    }
}

resource "aws_batch_job_queue" "staging-platform" {
    name = "staging-platform"
    compute_environments = ["${aws_batch_compute_environment.staging-platform.id}"]
    priority = 1
    state = "ENABLED"
}

resource "aws_batch_job_definition" "staging-platform-nightly-accrual-accounting" {
    name = "staging-platform-nightly-accrual-accounting"
    type = "container"
    container_properties = <<CONTAINER_PROPERTIES
{
    "image": "campspot/campspot:latest",
    "vcpus": 2,
    "memory": 2000,
    "command": [
        "node",
        "/campspot/server/bin/nightly-accrual-accounting.js"
    ],
    "volumes": [],
    "environment": [
        {
            "name": "NODE_ENV",
            "value": "staging"
        }
    ]
}
CONTAINER_PROPERTIES
    retry_strategy {
        attempts = 1
    }
    timeout {
        attempt_duration_seconds = 7200
    }
}

```

### Debug Output

running `terraform apply debug` doesn't work.

### Expected Behavior

Should allow me to specify the `timeout` property

### Actual Behavior

Gives me an error `Error: aws_batch_job_definition.staging-platform-nightly-accrual-accounting: : invalid or unknown key: timeout`

### Steps to Reproduce

1. `terraform apply`

### References

https://www.terraform.io/docs/providers/aws/r/batch_job_definition.html#timeout

**Comment by radeksimko** [comment]
Hi,
the new field was added in `v1.17.0` of this provider. Can you try upgrading and let us know if it solved the problem? Thanks.

https://github.com/terraform-providers/terraform-provider-aws/commit/1857746eb6c25780e97f627918a7461fef527a08

**Comment by bflad** [comment]
As mentioned above this was previously released in [version 1.17.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1170-may-02-2018) and has been available in all releases since. Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
